File: Benchmarking/30 Euler/euler6/euler6.cpp
//
//  main.cpp
//  Euler6
//
//  Created by Sonni on 19/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>
#include <math.h>

int diffe(unsigned int n)
{
    unsigned int sumFirst = 0;
    unsigned int sumSecond = 0;
    unsigned temp = 0;
    
    for (unsigned int i = 1; i <= n; i++)
        sumFirst += pow(i, 2);
    
    for (unsigned int j = 1; j <= n; j++)
        temp += j;
    
    sumSecond = pow(temp, 2);
    int res = sumSecond - sumFirst;
    if (res < 0)
        return res += res * -2;
    else
        return res;
}

int main(int argc, const char * argv[]) {
    std::cout << diffe(atoi(argv[1])) << std::endl;;
    return 0;
}

File: Benchmarking/noscaling/euler6/euler6.py
import sys

def diffe(n):
	sumDigitsSquare = 0
	sumNumberSquare = 0
	temp = 0
	for i in range(1, n+1):
		sumDigitsSquare += i**2
	for j in range(1, n+1):
		temp += j
	sumNumberSquare = temp**2
	res = sumNumberSquare - sumDigitsSquare
	if res < 0:
		return res * -2
	else:
		return res
	
limit = int(sys.argv[1])
diffe(limit)
